### ISSUE TITLE
question about IAST

### ISSUE THREAD

**Issue by gilbh** [Shreeshrii/tesstrain-Sanskrit-IAST#1]
I need to OCR a Sanskrit text in diacritics, preferably with Tesseract.

I am not sure how to use this IAST for Tesseract. Could you please provide instructions? I am using Tesseract on a Linux system, is there a iast.traineddata file that I can simply place inside /usr/shared/tesseract/4.00/tessdata? Any other way?

With thanks!

**Comment by Shreeshrii** [comment]
There are multiple traineddata files in this repo. All can be placed in /usr/shared/tesseract/4.00/tessdata and used. 
If you share a sample image I can check which one may be the best fit.

**Comment by gilbh** [comment]
Thank you.

I attach a few page scans, it's from Vettam Mani's Puranic Encyclopedia.

Also: what file extension do I place in tessdata? I thought it was only
.traineddata (which I couldn't find in the repo).

Thanks again
Gil

Gil Ben-Herut, PhD
Associate Professor
Director of Graduate Studies
Department of Religious Studies
University of South Florida
http://religious-studies.usf.edu/faculty/gbherut/

Check out my recent monograph *Śiva's Saints: The Origins of Devotion in
Kannada according to Harihara's *Ragaḷegaḷu
<https://global.oup.com/academic/product/sivas-saints-9780190878849?cc=us&lang=en&>and
the co-edited volume *Regional Communities of Devotion in South Asia:
Insiders, Outsiders, and Interlopers
<https://www.crcpress.com/Regional-Communities-of-Devotion-in-South-Asia-Insiders-Outsiders-and/Ben-Herut-Keune-Monius/p/book/9781138495838>*

**Comment by gilbh** [comment]
Sorry for confusion: there are trainedata files, but when I tried using one
the result had no diacritics...

Gil Ben-Herut, PhD
Associate Professor
Director of Graduate Studies
Department of Religious Studies
University of South Florida
http://religious-studies.usf.edu/faculty/gbherut/

Check out my recent monograph *Śiva's Saints: The Origins of Devotion in
Kannada according to Harihara's *Ragaḷegaḷu
<https://global.oup.com/academic/product/sivas-saints-9780190878849?cc=us&lang=en&>and
the co-edited volume *Regional Communities of Devotion in South Asia:
Insiders, Outsiders, and Interlopers
<https://www.crcpress.com/Regional-Communities-of-Devotion-in-South-Asia-Insiders-Outsiders-and/Ben-Herut-Keune-Monius/p/book/9781138495838>*

**Comment by gilbh** [comment]
The scans are attached here.
[test_Mani_Purana.pdf](https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/files/6015723/test_Mani_Purana.pdf)

**Comment by Shreeshrii** [comment]
[test_Mani_Purana-DevaLayer.txt](https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/files/6015769/test_Mani_Purana-DevaLayer.txt)

I tried with one of the traineddata. Result is not perfect , but it has diacritics.

**Comment by Shreeshrii** [comment]
You can use -l eng+DevaLayer...  for better results.

[test_Mani_Purana-eng-DevaLayer.txt](https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/files/6015918/test_Mani_Purana-eng-DevaLayer.txt)

**Comment by gilbh** [comment]
Thank you. The first txt file you sent has AVATĀRA, which is exactly what I
need, but perhaps I am not calling Tesseract with the right parameters?
I am doing it via Python:
```
pytesseract.image_to_string(image_for_tesseract, lang='iast')
```
whereas 'iast' is the file I downloaded from your repo (
San_0.937_485114_1579900.traineddata
<https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/blob/main/tessdata_best/San_0.937_485114_1579900.traineddata>
) and renamed as
iast.traineddata

What am I doing wrong?

Thanks
Gil


Gil Ben-Herut, PhD
Associate Professor
Director of Graduate Studies
Department of Religious Studies
University of South Florida
http://religious-studies.usf.edu/faculty/gbherut/

Check out my recent monograph *Śiva's Saints: The Origins of Devotion in
Kannada according to Harihara's *Ragaḷegaḷu
<https://global.oup.com/academic/product/sivas-saints-9780190878849?cc=us&lang=en&>and
the co-edited volume *Regional Communities of Devotion in South Asia:
Insiders, Outsiders, and Interlopers
<https://www.crcpress.com/Regional-Communities-of-Devotion-in-South-Asia-Insiders-Outsiders-and/Ben-Herut-Keune-Monius/p/book/9781138495838>*

**Comment by Shreeshrii** [comment]
The complete Puranic Encyclopedia has already been digitized. Please see https://www.sanskrit-lexicon.uni-koeln.de/scans/PEScan/2020/web/index.php

**Comment by gilbh** [comment]
Thanks. I was familiar with their Monier-Williams but did not know they had
digitized Pur Enc as well!

The reason I am interested in OCR this text is to add it to my digitalRoses
website. You can check it out here:
https://digitalroses.net/grav/
and also:
https://digitalroses.net/grav/examples

But I of course have a larger interest in OCR for diacritics given my
research (please see links in signature below). I am sorry to say but I am
still at lost about utilizing your trainddata and would greatly appreciate
your help. I am sure I am not very far from succeeding, following my
description below:


Via Python:
```
pytesseract.image_to_string(image_for_tesseract, lang='iast')
```
whereas 'iast' is the file I downloaded from your repo (
San_0.937_485114_1579900.traineddata
<https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/blob/main/tessdata_best/San_0.937_485114_1579900.traineddata>
) and renamed as
iast.traineddata

What am I doing wrong?


this and would greatly
Gil Ben-Herut, PhD
Associate Professor
Director of Graduate Studies
Department of Religious Studies
University of South Florida
http://religious-studies.usf.edu/faculty/gbherut/

Check out my recent monograph *Śiva's Saints: The Origins of Devotion in
Kannada according to Harihara's *Ragaḷegaḷu
<https://global.oup.com/academic/product/sivas-saints-9780190878849?cc=us&lang=en&>and
the co-edited volume *Regional Communities of Devotion in South Asia:
Insiders, Outsiders, and Interlopers
<https://www.crcpress.com/Regional-Communities-of-Devotion-in-South-Asia-Insiders-Outsiders-and/Ben-Herut-Keune-Monius/p/book/9781138495838>*

**Comment by gilbh** [comment]
By way of following up:
Just managed to get very good results with 

DevaLayer-1017.traineddata 

This is great. Thanks a lot for your help!

**Comment by Shreeshrii** [comment]
I have uploaded [a new traineddata file](https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/blob/main/tessdata_fast/PuranaFinetune-210224.traineddata)  which should give better results for the  Pur Enc.

**Comment by gilbh** [comment]
This is fantastic! 

I am running it right now (is it slower than before? Takes a good chunk of time, perhaps twenty minutes already and only covered 50%...). 

You can already see the results of the previous traineddata file (DevaLayer-1017.traineddata), which are very good, here:
https://indianlit.digitalroses.net

Let me know what you think.

Best

**Comment by Shreeshrii** [comment]
>You can already see the results of the previous traineddata file (DevaLayer-1017.traineddata), which are very good, here:
https://indianlit.digitalroses.net

Great! I see now, you only needed the HEADWORD on page. The problem with `e/c` is there only for full text OCR.

>is it slower than before? Takes a good chunk of time

It is possible, since I had added a large wordlist in the hope of improving results. Or, maybe you are comparing speed of tessdata_best vs tessdata_fast?

Based on this, I have removed tessdata_best (which is slower) from the repo as also the other two traineddatas. Hopefully this will make it easier for others.
